tests/integration/oauth_adapter_test.py

```diff
@@ -49,17 +49,21 @@ def authorize_with_providers(cls, oauth_adapters):
         redirect_uri = "http://local.broadinstitute.org/#fence-callback"
         state = "abc123"
         authz_responses = {}
+        # Whether this test is being run manually or for by automated integration.
+        # The integration run uses a fake provider. The manual run allows testing against real providers with real
+        # authentication. There's not a good way to run integration tests against real providers, so we have this
+        # flag to manually change the tests.
+        manual_run = False
         for provider, oauth_adapter in oauth_adapters.iteritems():
             authz_url = oauth_adapter.build_authz_url(scopes,
                                                       redirect_uri,
                                                       state,
-                                                      extra_authz_url_params={"idp": "google"})
+                                                      extra_authz_url_params={"idp": "google"} if manual_run else {"foo": "bar"})
             print("Please go to %s to authorize access: %s" % (provider, authz_url))
             print("YOU WILL BE REDIRECTED TO %s WHICH WILL PROBABLY UNREACHABLE -- THIS IS EXPECTED!" % redirect_uri)
             print("Please copy/paste the \"code\" parameter from the resulting URL: ")
             sys.stdout.flush()
-            auth_code = sys.stdin.readline().strip()
-            # auth_code = "X"
+            auth_code = sys.stdin.readline().strip() if manual_run else "X"
```

Comment:
python ternary syntax takes some getting used to :\

Is this line needed just so that in an automated run, it doesn't wait for stdin? Because otherwise won't it fail with an auth code of "X"? / Is this test ever run in automated tests?

Reply:
It is needed so that the automated run will pass, otherwise we get an error, like this branch has now: https://circleci.com/gh/DataBiosphere/bond/593?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Reply:
makes sense -- thanks!